.pre-commit-config.yaml

```diff
@@ -0,0 +1,34 @@
+repos:
```

Comment:
I remember seeing a comment that keeping pre-commit up to date can be difficult. Well, we can automate that part too https://github.com/pydanny/cookiecutter-django/blob/master/.github/workflows/pre-commit-autoupdate.yml

Reply:
This is the setting I use in one of my projects.

```yaml
  - repo: https://github.com/browniebroke/django-codemod
    # django-codemod is a tool to automatically
    # fix Django deprecations.
    # Any tag/version (>=1.5.7):
    # https://github.com/browniebroke/django-codemod/tags
    #
    # Automatic execution of this hook has been disabled
    # because it is a bit slow, but its configuration is
    # kept here, so it can be easily run:
    #
    # pre-commit run --hook-stage manual djcodemod
    # or
    # pre-commit run -a --hook-stage manual djcodemod
    rev: v1.5.7
    hooks:
      - id: djcodemod
        stages: [manual]
        args: [
            'run',
            '--deprecated-in', '1.9',
            '--deprecated-in', '1.10',
            '--deprecated-in', '1.11',
            '--deprecated-in', '2.0',
            '--deprecated-in', '2.1',
            '--deprecated-in', '2.2',
#            '--deprecated-in', '3.0',
#            '--deprecated-in', '3.1',
        ]
  - repo: https://github.com/asottile/pyupgrade
    rev: v2.23.1
    hooks:
      - id: pyupgrade
        args: ['--py3-plus', '--py39-plus']
  - repo: https://github.com/pre-commit/pre-commit-hooks
    rev: v4.0.1
    hooks:
    -   id: fix-byte-order-marker
    -   id: check-case-conflict
    -   id: check-merge-conflict
    -   id: check-symlinks
    -   id: debug-statements
    -   id: detect-private-key
    -   id: double-quote-string-fixer
    -   id: mixed-line-ending
    -   id: no-commit-to-branch
        args: [--branch, main, --branch, master]
  - repo: https://github.com/PyCQA/flake8
    rev: 3.9.2
    hooks:
    -   id: flake8
```

Reply:
I'm still not sure about `pre-commit` =/

Reply:
Pre-commit settings have been removed.